### ISSUE TITLE
R6034 runtime error when registering pythoncom component.

### ISSUE THREAD

**Issue by jazzatar** [mhammond/pywin32#1797]
* Expected behavior and actual behavior.
### Some background
We are upgrading a 32-bit software project from Python 2.5.4 pywin32-214 to use the last release of Python 2.7.18 (before we migrate to Python 3).

Our software registers a COM component that provides the Python interpreter and pywin32 in order to add automation within our software package. It adds this support through the use of #include <Python.h> and #include <pythoncom.h>.

### Issue
When attempting to register our component via regsvr32, we get the R6034 runtime error (same as #566, but through different door). If I were to guess, this is related to the addition of PythonCOMLoader to change contexts for loading msvcr90.dll.

![r6034](https://user-images.githubusercontent.com/15916524/141849977-66168e84-076c-4234-ae28-ed1055b34ff5.png)

* Steps to reproduce the problem.
I have not yet created an example project to show this issue from Visual Studio, but a command prompt will show the issue as well:

> cd C:\Python27\Lib\site-packages\pywin32_system32
> regsvr32 pythoncom27.dll
< R6034 runtime error occurs >

* Version of Python and pywin32
Python 2.7.18 and pywin32-228

**Comment by mhammond** [comment]
> regsvr32 pythoncom27.dll

pythoncom27.dll doesn't implement any COM objects, so this doesn't really make sense. But yeah, this does sound like the problem pythoncomloader solves, and all COM objects which are registered should use that loader instead of pythoncom.dll.

**Comment by mhammond** [comment]
(and FYI, that loader doesn't existing in 3.x, as the compilers used to build later Python versions don't have the same issue - so this will go away entirely when you move to 3.x)

**Comment by jazzatar** [comment]
> > regsvr32 pythoncom27.dll
> 
> pythoncom27.dll doesn't implement any COM objects, so this doesn't really make sense.

That is totally fair and true, but our module does implement a few COM objects, and it has direct dependencies on python27.dll, and pythoncom27.dll. While the command is silly to call on pythoncom27.dll, this is the same error that occurs upon calling regsvr32 on our module.

[regsvr32debuggeroutput.txt](https://github.com/mhammond/pywin32/files/7542457/regsvr32debuggeroutput.txt)

You can actually see the attempt to load the Python DLLs fail under the debugger.

I'm still trying to sort out how to bring in the pythoncomloader logic into our COM component to see if that fixes it.

**Comment by mhammond** [comment]
> I'm still trying to sort out how to bring in the pythoncomloader logic into our COM component to see if that fixes it.

That might be tricky in your case - pythoncomloader.dll is going to want to find pythoncomXX.dll, but it sounds like you need it to find your dll?

The other option might be to ensure that your DLL has an appropriate "manifest" - see 097a4261eec790b29ec72257a570512a0ee9ced5 where I stopped trying to build that DLL, but it shows the linker ends up creating a manifest file, then we needed to run the `mt` tool to stamp the final DLL with it. It's been a few years since I had to work out this manifest cruft and I can't remember many details - it's no surprise MS killed it entirely in newer compilers.

**Comment by jazzatar** [comment]
Voila! Thanks for the idea. I was able to pull the manifest out of python27.dll: 
`mt -inputresource:python27.dll;#2 -out:python27.manifest`

And stuff that into our component. That worked! Attaching the manifest I added in case it helps someone in the future with this strange use case (added .txt to the end to make github happy)

[ShellScript.dll.manifest.txt](https://github.com/mhammond/pywin32/files/7542594/ShellScript.dll.manifest.txt)

This is workable for us.
